File: lib/rin/io/events/CharacterEvent.cpp
#include "rin/io/events/CharacterEvent.h"

namespace rin::io
{
    
    CharacterEvent::CharacterEvent(Window* inWindow, const char inCharacter, const Flags<InputModifier>& inModifiers) : Event(inWindow), character(inCharacter), modifiers(inModifiers)
    {
        
    }
}

File: Rin.Engine/Views/Events/CharacterEvent.cs
using Rin.Engine.Graphics.Windows;
using Rin.Engine.Views.Graphics;

namespace Rin.Engine.Views.Events;

public class CharacterEvent(Surface surface, char character, InputModifier mods) : Event(surface)
{
    public char Character = character;
    public InputModifier Mods = mods;
}